scanner/src/main/kotlin/Scanner.kt

```diff
@@ -617,12 +617,16 @@ class Scanner(
 
     private fun postProcessScanSummary(summary: ScanSummary, downloadDir: File) =
         summary.copy(
-            licenseFindings = summary.licenseFindings.mapTo(mutableSetOf()) {
-                val licenseString = it.license.toString()
-                it.copy(
-                    license = licenseString.mapLicense(scannerConfig.detectedLicenseMapping).toSpdx(),
-                    location = it.location.withRelativePath(downloadDir)
-                )
+            licenseFindings = summary.licenseFindings.flatMapTo(mutableSetOf()) {
```

Comment:
Am I right this is done before writing the Summary to the storages? 
If so, I'm strongly vote not doing it here, but after the summary gets pulled out from storage, so that one can change this configuration parameter without invalidating the scan cache.

Reply:
My goal was to not change semantics compared to before. Both the original `parseLicenseExpressions` and now `decomposeLicenses` happen at the same time in this regard, i.e. before writing to the scan storage.

Reply:
And BTW, I'd more more than happy to drop `parseLicenseExpressions` completely without any substituting feature. I just added `decomposeLicenses` as I feared there would be too much push-back for this PR if I do not provide a substituting feature.

Reply:
@oss-review-toolkit/core-devs any feedback on whether `parseLicenseExpressions` for ScanCode can be dropped without introducing `decomposeLicenses` for all scanners?

Reply:
BTW, I just realized there is https://github.com/oss-review-toolkit/ort/issues/5017, which IIUC means that even with `parseLicenseExpressions = true` at least the web-app report doe snot show license expressions. Which to me is more more reason that `parseLicenseExpressions` can safely be removed and make it default to `true` which changing behavior.

Reply:
> can safely be removed and make it default to `true` which changing behavior.

So, the target behavior is as follows, right? 
1. Each compound license expression returned by ScanCode will remain a compound license expression in   
    ScanSumamry?
2. The ScanCode e2e integration won't decompose any license expressions

Reply:
That's my proposal, yes. (Note the "which" -> "without" typo I made in the text you quoted.) That is, depending on the discussion outcome, I'd simply drop the last commit form this PR.

Reply:
IIRC the main reason to introduce the `parseLicenseExpressions` flag was that at the time we were concerned that enabling it at HERE would require a large rework of the existing package configurations and might require an update of the rules, both of which we did not have capacity for back then (see  04584d2). However, I wonder if nowadays anyone is actually setting this flag to `false`.

Reply:
> However, I wonder if nowadays anyone is actually setting this flag to `false`.

Well, it's `false` by default still (`isTrue()` returns `false` for `null`, i.e. if the option is unset).